CS410_Assignment2/src/SensorApplication/sensor/SensorAdapter.java

```diff
@@ -0,0 +1,34 @@
+/* SensorAdapter is an abstract class that contains all the fields and methods shared 
+ * by all of its derived classes: TemperatureSensorAdapter, PressureSensorAdapter, RadiationSensorAdapter.
+ */
+
+package sensor;
+
+import java.awt.Color;
+
+public abstract class SensorAdapter implements Sensor{
+
+	@Override
+	public String getType() {
+		return type;
+	}
+	
+	@Override
+	public Color getStatusColor() {
+		if(status == "OK")
+			return Color.GREEN;
+		else if(status == "CRITICAL")
+			return Color.YELLOW;
+		else
+			return Color.RED;
+	}
+	
+	@Override
+	public String getStatusInfo() {
+		return status + "-->" + value;
+	}
+	
+	protected String type;
+	protected String status;
+	protected double value;
+}

```

Comment:
This class is mostly redundant, because it simply passes the data from [sensorName]Adapter to the Sensor interface.  The functions `getStatusInfo()` and `getStatusColor()` should probably be handled by the GUI, because they simply format the raw data coming from the sensors.